src/components/actions/dev_tools/routes.ts

```diff
@@ -0,0 +1,264 @@
+    trigger: Trigger.DevTest,
+    table: 'application',
+    record_id: appId,
+    applicationDataOverride,
+  }
+
+  switch (triggerFull) {
+    case 'DEV_TEST':
+      actionsOutput = await processTrigger(triggerPayload)
+      finalApplicationData = await getApplicationData({ applicationId })
+      break
+
+    case 'ON_APPLICATION_CREATE':
+      // To-do: actually create an "is_config" application -- currently relies
+      // on one already existing
+      triggerPayload.trigger = Trigger.OnApplicationCreate
+      actionsOutput = await processTrigger(triggerPayload)
+      finalApplicationData = await getApplicationData({ applicationId })
+      break
+
+    case 'ON_APPLICATION_RESTART':
+      triggerPayload.trigger = Trigger.OnApplicationRestart
+      actionsOutput = await processTrigger(triggerPayload)
+      finalApplicationData = await getApplicationData({ applicationId })
+      break
+
+    case 'ON_APPLICATION_SUBMIT':
+      triggerPayload.trigger = Trigger.OnApplicationSubmit
+      actionsOutput = await processTrigger(triggerPayload)
+      finalApplicationData = await getApplicationData({ applicationId })
+      break
+
+    case 'ON_EXTEND':
+      triggerPayload.trigger = Trigger.OnExtend
+      actionsOutput = await processTrigger(triggerPayload)
+      finalApplicationData = await getApplicationData({ applicationId })
+      break
+
+    case 'ON_SCHEDULE':
+      if (!eventCode) return reply.send('eventCode required')
+      const { id, data } = await db.getScheduledEvent(applicationId, eventCode)
+      console.log(id)
+      triggerPayload.trigger = Trigger.OnSchedule
+      triggerPayload.table = 'trigger_schedule'
+      triggerPayload.record_id = id
+      triggerPayload.data = data
+      actionsOutput = await processTrigger(triggerPayload)
+      finalApplicationData = await getApplicationData({ applicationId })
+      break
+
+    case 'ON_PREVIEW':
+      const revId = reviewId ?? (await getRandomReviewId(applicationId))
+      triggerPayload.trigger = Trigger.OnPreview
+      triggerPayload.table = 'review'
+      triggerPayload.record_id = revId
+      triggerPayload.applicationDataOverride = {
+        ...applicationDataOverride,
+        reviewData: {
+          latestDecision: {
+            decision: decision ?? Decision.Conform,
+            comment: comment ?? 'Test comment',
+          },
+        },
+      }
+
+      actionsOutput = await processTrigger(triggerPayload)
+      finalApplicationData = await getApplicationData({ applicationId, reviewId: revId })
+      break
+
+    case 'ON_REVIEW_ASSIGN':
+      // If no assignment id, select one at random (using latest stage and level) and assign all possible sections to it
+      {
+        const assignment = assignmentId
+          ? await db.getSingleReviewAssignment(assignmentId)
+          : await selectRandomReviewAssignment(applicationId, sectionCodes)
+        if (!assignment.allowedSections) assignment.allowedSections = sectionCodes
```

Comment:
This seems as it's already done inside ` selectRandomReviewAssignment`

Reply:
The "allowed sections" field can be NULL, in which case we treat it as "allow all". So this just does that here -- puts all the section codes in if null.